### PR TITLE
Fix: Variable name generation ignoring custom size labels [ED-23051]

### PR DESCRIPTION
**Pull request by Ntnelbaba**
## Summary
- Fixed `getDefaultName` in `variables-manager-create-menu.tsx` to correctly account for **all** size sub-types when generating a new variable name
- The previous filter (`variable.type === type`) only checked variables of the exact same type key (e.g. `global-size-variable`), silently ignoring custom size variables (`global-custom-size-variable`) that share the same `size-N` label pattern — leading to duplicate names being generated
- Replaced the type-based filter with a regex pattern match (`^size-(\d+)$`, case-insensitive) across **all** variables, so both standard and custom size variables are counted when finding the next available number
- Removed the now-unused `type` parameter from the function signature
- Added a dedicated unit test file with 6 scenarios, including the specific bug regression case

## Test plan
- [ ] Open the Variables Manager and create several custom size variables (e.g. `size-1`, `size-2`)
- [ ] Create a new standard size variable and confirm the generated name is `size-3`, not `size-1` or `size-2`
- [ ] Verify the same behaviour when gaps exist (e.g. only `size-1` and `size-3` exist → new variable gets `size-2`)
- [ ] Run unit tests: `cd /path/to/elementor/packages && npx jest packages/core/editor-variables --no-coverage` — all 194 tests should pass

## Jira
[ED-23051](https://elementor.atlassian.net/browse/ED-23051)

[ED-23051]: https://elementor.atlassian.net/browse/ED-23051?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### ✨ PR Description
Purpose: Fix variable name generation logic to properly detect and avoid conflicts with custom size label variables when creating new variables.

Main changes:
- Refactored getDefaultName() to use regex pattern matching instead of type-based filtering, preventing naming conflicts with custom variables
- Removed type parameter from getDefaultName() function signature, making name collision detection type-agnostic across all variable types
- Added comprehensive test suite covering edge cases including custom variable types, case-insensitivity, and gap-filling in numbering

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: packages/packages/core/editor-variables/src/components/variables-manager/variables-manager-create-menu.tsx [hunk 1]
```diff
@@ -139,18 +139,20 @@ const MenuOption = ( {
 	);
 };
 
-const getDefaultName = ( variables: TVariablesList, type: string, baseName: string ) => {
-	const existingNames = Object.values( variables )
-		.filter( ( variable ) => variable.type === type )
-		.map( ( variable ) => variable.label );
+export const getDefaultName = ( variables: TVariablesList, baseName: string ) => {
+	const pattern = new RegExp( `^${ baseName }-(\\d+)$`, 'i' );
```
[the remark this answers is not in the document]
Comment (mike-elementor) [reply packages/packages/core/editor-variables/src/components/variables-manager/variables-manager-create-menu.tsx:143]: why can't we just count total number of "size" / "color" / "font" and slap it to the end?
so like:
```
section-width[size]
v-auto[custom-size]

reTry=500;
do {
  newLabel = `${ type }-{ ++counter }`;
  if (!lookup.has(newLabel) return newLabel;
} while(0 < --reTry);
```

there will not be a need to use user-defined values inside the code... sort-of

**Review by mike-elementor — APPROVED** [review]
